### DISCUSSION TITLE
outer fixture not run when running nested test

### DISCUSSION THREAD

**Discussion by stefanseefeld** [pytest-dev/pytest#8917] [Q&A]
Hello,
I have a test suite with a couple of subdirectories, each of which adds sub-directory specific fixtures, in addition to the toplevel `conftest.py` providing generic fixtures used by all tests.

How can I invoke a specific test from a sub-directory, while still using fixtures defined in "outer" `conftest.py` files ?
Assume a structure like 
```
tests/
    conftest.py
    a/
        conftest.py
        test_1.py
        test_2.py
    b/
        conftest.py
        test_3.py
```
With the above, if I run `pytest tests/b/test_3.py`, fixtures from `tests/conftest.py aren't seen, so I get an internal error if those tests require such fixtures. What is the expected way to run individual tests with the above testsuite structure ?

**Comment by RonnyPfannschmidt** [comment]
it should work, if it  does in fact not, thats a bug we need to fix

do you by chance have a minimal reproducer?

**Reply by stefanseefeld** [reply]
Actually, I mis-spoke: it's not a missing fixture I'm observing, but a top-level `pytest_addoption()` call isn't being executed, so a command-line option I added isn't being found by the nested test.

**Reply by RonnyPfannschmidt** [reply]
then its possibly an issue/edgecses with early conftest finding (pytest doesnt try to eagerly find **all** conftests before a testrun)

**Reply by stefanseefeld** [reply]
Any idea how I can work around that ? Or better, debug and fix it ? :-)

**Reply by RonnyPfannschmidt** [reply]
unfortunately not off-hand, a minimal reproducer would be a good starter
